src/mpid/ch4/src/ch4r_callbacks.c

```diff
@@ -263,7 +264,7 @@ static int do_send_target(void **data, size_t * p_data_sz, int *is_contig,
     MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_DO_SEND_TARGET);
 
     *target_cmpl_cb = recv_target_cmpl_cb;
-    MPIDIG_REQUEST(rreq, req->seq_no) = OPA_fetch_and_add_int(&MPIDI_global.nxt_seq_no, 1);
+    MPIDIG_REQUEST(rreq, req->seq_no) = MPL_atomic_fetch_add_uint64(&MPIDI_global.nxt_seq_no, 1);
```

Comment:
Since by far the most of such usages are to add or subtract 1, I am tempted to introduce wrapper macro `MPL_atomic_inc_uint64` and `MPL_atomic_dec_uint64`. What do you think?

Reply:
Yes, makes sense to me. I suppose OPA had these for the same reason.

Reply:
My concern about `inc` is that we don't know whether this operation is fetch-and-add or add-and-fetch (=whether its return value is a value before addition or one after addition). If `void`-type `inc` and `dec` are useful, we can make wrappers. If not, I like fetch-add because its behavior is clear.

Reply:
Just follow conventional `c++` semantics, which will be a "fetch-and-op", always.

Reply:
Both `i++` and `++i` are accepted in C/C++. I believe it is worth writing a little bit more (a longer function name (`MPL_atomic_fetch_add_uint64`) plus write a constant value `1`) to avoid any confusion.

Reply:
In the case of `i++`, there is no telling which value it returns just from the surface, we learn them as convention and convention makes code more readable.

Anyway, if you insist, I am OK with it. 👍

Reply:
I don't mind new functions if the convention is well established, but in my understanding, atomic increment does not have a commonly shared operational semantics. We should basically use primitive atomic operations such as load/store/fetch-add/add-fetch/CAS/test-and-set/swap, which are commonly shared in the community.